src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -413,13 +413,28 @@ private static void LogIfVerbose(GenerateAzurePayloadSettings settings, string m
         /// </summary>
         /// <param name="settings">The Azure settings.</param>
         /// <returns>A QIR submitter.</returns>
-        private static IQirSubmitter? QirSubmitter(AzureSettings settings) => settings.Target switch
+        private static IQirSubmitter? QirSubmitter(AzureSettings settings)
         {
-            null => null,
-            NoOpQirSubmitter.Target => new NoOpQirSubmitter(),
-            NoOpSubmitter.Target => new NoOpSubmitter(),
-            _ => SubmitterFactory.QirSubmitter(settings.Target, settings.CreateWorkspace(), settings.Storage)
-        };
+            if (settings.Target is null)
+            {
+                return null;
+            }
+
+            // If a configuration file is provided, create a submitter using it.
+            if (!(settings.SubmitConfigFile is null))
```

Comment:
(a note for oneself)
```suggestion
            if (settings.SubmitConfigFile is not null)
```
([example](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/safely-cast-using-pattern-matching-is-and-as-operators))

Reply:
src/Simulation/EntryPointDriver/Azure/Azure.cs(424,46): error CS8400: Feature 'not pattern' is not available in C# 8.0. Please use language version 9.0 or greater.